query/query.go

```diff
@@ -231,6 +231,23 @@ func (q *BranchesRepos) String() string {
 	return sb.String()
 }
 
+// NewSingleRepoIds is a helper for creating a RepoIds which
+// searches only the matched repos.
+func NewRepoIds(ids ...uint32) *RepoIds {
+	return &RepoIds{Repos: roaring.BitmapOf(ids...)}
+}
+
+func (q *RepoIds) String() string {
+	var sb strings.Builder
+
+	sb.WriteString("(repoids ")
+
+	sb.WriteString(q.Repos.String())
```

Comment:
I notice in `ReposBranches` we optimize to only printing the count of repos being filtered. I assume that these strings are primarily to aid in debugging and logging visibility so the choice here may be a little arbitrary. Should we opt to just print the count here if it's above a certain number (1) like we do with `ReposBranches`? Maybe this is advisable to reduce the risk of massive log lines somewhere?

Reply:
Yes we should do this. For example this string can end up in traces and logs. A query string with 10k repos makes those logs useless / spans too large.